internal/services/monitor/monitor_diagnostic_setting_resource.go

```diff
@@ -240,8 +296,8 @@ func resourceMonitorDiagnosticSettingCreateUpdate(d *pluginsdk.ResourceData, met
 		}
 	}
 
-	if !valid {
-		return fmt.Errorf("At least one `log` or `metric` must be enabled")
+	if !valid && !hasEnabledLogs {
+		return fmt.Errorf("at least one `log` or `metric` must be enabled")
```

Comment:
we can update this to the _type_ rather than the property here?

```suggestion
		return fmt.Errorf("at least one type of Log or Metric must be enabled")
```